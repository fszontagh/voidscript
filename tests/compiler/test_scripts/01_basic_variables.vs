// Test basic variable declarations and assignments
int $a = 10;
string $b = "Hello";
bool $c = true;
float $d = 3.14;

printnl("Integer: ", $a);
printnl("String: ", $b);
printnl("Boolean: ", $c);
printnl("Float: ", $d);

// Variable reassignment
$a = 20;
$b = "World";
$c = false;
$d = 2.71;

printnl("Updated integer: ", $a);
printnl("Updated string: ", $b);
printnl("Updated boolean: ", $c);
printnl("Updated float: ", $d);