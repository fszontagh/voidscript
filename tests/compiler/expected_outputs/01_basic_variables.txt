Integer: 10
String: Hello
Boolean: true
Float: 3.14
Updated integer: 20
Updated string: World
Updated boolean: false
Updated float: 2.71
